samples/subsys/settings/sample.yaml

```diff
@@ -13,6 +13,7 @@ tests:
       - nrf54l15dk/nrf54l05/cpuapp
       - nrf54l15dk/nrf54l10/cpuapp
       - nrf54l15dk/nrf54l15/cpuapp
+      - nrf54l20pdk/nrf54l20/cpuapp
```

Comment:
```suggestion
```

There is nothing to test, it's a sample (will follow up with a PR to clean up the rest of the list if I don't forget)
See https://docs.zephyrproject.org/latest/samples/sample_definition_and_criteria.html#twister-should-be-able-to-build-every-sample

Reply:
On that page/chapter there is:
"Use platform_allow to limit test to the platforms the sample was actually verified on. Those platforms should be listed in the sample’s README file."

I believe this sample doesn't work out-of-the box on every platform. At least, nRF boards require some KConfig tweaks. However, I see in the 'boards' directory overlay files for boards that are not listed under platform_allow.

I will add commit that removes platform_allow from this sample to check how CI will react.
TBC

Reply:
platform_allow must stay. Otherwise integration CI is failing:
```
INFO    - The following issues were found (showing the top 10 items):
INFO    - 1) sample.subsys.settings on m2gl025_miv/miv failed (Timeout)
INFO    - 2) sample.subsys.settings on mps2/an385 failed (Timeout)
INFO    - 3) sample.subsys.settings on mps2/an521/cpu0 failed (Timeout)
INFO    - 4) sample.subsys.settings on qemu_arc/qemu_arc_em failed (Timeout)
INFO    - 5) sample.subsys.settings on qemu_arc/qemu_arc_hs failed (Timeout)
INFO    - 6) sample.subsys.settings on qemu_arc/qemu_arc_hs/xip failed (Timeout)
INFO    - 7) sample.subsys.settings on qemu_arc/qemu_arc_hs5x failed (Timeout)
INFO    - 8) sample.subsys.settings on qemu_arc/qemu_arc_hs6x failed (Timeout)
INFO    - 9) sample.subsys.settings on qemu_cortex_a53/qemu_cortex_a53 failed (Timeout)
INFO    - 10) sample.subsys.settings on qemu_cortex_a53/qemu_cortex_a53/smp failed (Timeout)
```

Reply:
I never asked you to remove `platform_allow` altogether, just the line you'd added :) see my request for change.

Reply:
If I remove that line, Twister will skip this test on nrf54l20pdk because it's not in platform_allow.

So, what do You mean by "will follow up with a PR to clean up the rest of the list if I don't forget"?

Reply:
> If I remove that line, Twister will skip this test/sample on nrf54l20pdk because it's not in platform_allow.

Yes, that's the whole point! :) There is no need for Twister to go through each sample and test them against dozen of platforms, this just doesn't scale in terms of project infrastructure. 

> So, what do You mean by "will follow up with a PR to clean up the rest of the list if I don't forget"?

I would like you to not add the new line, and by follow-up I mean I will clean up the other lines so that people stop unecessarily adding to this list and to keep Twister "testing" to the strict minimum, i.e test that the sample builds fine and runs fine on e.g native sim.

Reply:
> > If I remove that line, Twister will skip this test/sample on nrf54l20pdk because it's not in platform_allow.
> 
> Yes, that's the whole point! :) There is no need for Twister to go through each sample and test them against dozen of platforms, this just doesn't scale in terms of project infrastructure.

I was referring to Nordic's internal CI that executes on-target Twister tests. Currently, that's the only place where nrf54L20 PDK can be tested.
Based on Twister results we calculate metrics that "give overview" how well each target is tested, SW maturity level, etc. (this statement is an oversimplification).

> > So, what do You mean by "will follow up with a PR to clean up the rest of the list if I don't forget"?
> 
> I would like you to not add the new line, and by follow-up I mean I will clean up the other lines so that people stop unecessarily adding to this list and to keep Twister "testing" to the strict minimum, i.e test that the sample builds fine and runs fine on e.g native sim.

Nordic Management would like to have similar test coverage on nrf54L20 as on other platforms. As a result, I was given task to enable Twister tests on that target (in Nordic's internal CI but this also enables testing in the "upstream CI").
I need to discuss this with my boss because our expectations are conflicting. Unfortunately, he's out of office until January 2025.

So, I guess it's not as simple as run "upstream Twister" with options:
`-T TESTSUITE_ROOT, --testsuite-root TESTSUITE_ROOT` to select which tests to execute and/or
`-P EXCLUDE_PLATFORM, --exclude-platform EXCLUDE_PLATFORM` to skip testing on given platform and/or
`-p PLATFORM, --platform PLATFORM` to run tests on f.e. native sim only.

> I will clean up the other lines so that people stop unecessarily adding to this list and to keep Twister "testing" to the strict minimum

Are other manufacturers aware that it's planned to limit "upstream Twister" scope in a way that silently limits test scope in "private Twister runs"?